src/cpp/src/sequence_group.hpp

```diff
@@ -477,7 +477,7 @@ class SequenceGroup {
         }
         // For beam search streaming is not available, so we notify only upon finishing
         if(m_sampling_params.is_beam_search()) {
-            if (has_finished() || out_of_memory()) {
+            if (has_finished() || out_of_memory() || handle_dropped()) {
```

Comment:
Don't we need this additional check also for greedy and multinomial?

Reply:
Yes, we need. Fixed that now